src/sources/commondatamodel.jl

```diff
@@ -31,24 +31,38 @@ const CDM_STANDARD_NAME_MAP = Dict(
     "time" => Ti,
 )
 
+# CFVariable is imblemented again here because the one
+# in CommonDataModel is not DiskArrays.jl compatible.
+# TODO move this code to CommonDataModel.jl
 struct CFVariable{T,N,TV,TA,TSA} <: CDM.AbstractVariable{T,N}
     var::CDM.CFVariable{T,N,TV,TA,TSA}
 end
 @implement_diskarray CFVariable
 
+# Base methods
 Base.parent(A::CFVariable) = A.var
+Base.getindex(os::OpenStack{<:CDMsource}, key::Symbol) = CFVariable(dataset(os)[key])
 
+# DiskArrays.jl methods
 function DiskArrays.readblock!(A::CFVariable, aout, i::AbstractUnitRange...)
-    aout[i...] = getindex(parent(A), i...)
+    aout .= getindex(parent(A), i...)
 end
-
 function DiskArrays.writeblock!(A::CFVariable, data, i::AbstractUnitRange...)
     setindex!(parent(A), data, i...)
     return data
 end
 _open(f, ::Type{<:CDMsource}, var::CFVariable; kw...) = cleanreturn(f(var))
 
-# Methods from CommonDataModel
+# We have to dig down to find the chunks as they are not immplemented
+# properly in the source packages, but they are in their internal objects.
+# We just make it work here even though it doesn't work in NCDatasets.jl or GRIBDatasets.jl
+DiskArrays.eachchunk(var::CFVariable) = _get_eachchunk(var.var.var)
+DiskArrays.haschunks(var::CFVariable) = _get_haschunks(var.var.var)
+
+function _get_eachchunk end
+function _get_haschunks end
```

Comment:
@tcarion @Alexander-Barth this is where the hack is called to dig into the inner array that knows the actual chunk pattern. With the current design of NCDatasets.jl/GRIBDatasets.jl/CommonDataModel.jl anyone wanting chunking to work with CFVariable will need to do a hack like this.

(Note this is the `Rasters.CFVariable`, not type piracy)